libs/sfm/ba_linear_solver.cc

```diff
@@ -72,7 +72,12 @@ namespace
             math::Matrix<double, 3, 3> rot;
             for (int i = 0; i < 9; ++i)
                 rot[i] = *(iter++);
-            rot = math::matrix_inverse(rot);
+
+            double det = math::matrix_determinant(rot);
+            if (std::abs(det) < MATH_DBL_EPS)
```

Comment:
There is a math define which is more efficient that `std::abs`, please use `MATH_EPSILON_EQ` instead.